filebeat/prospector/prospector.go

```diff
@@ -214,7 +223,22 @@ func (p *Prospector) updateState(event *input.Event) error {
 	event.Module = p.config.Module
 	event.Fileset = p.config.Fileset
 
-	ok := p.outlet.OnEvent(event)
+	eventHolder := event.GetEventHolder()
+	//run the filters before sending to
+	if event.Bytes > 0 {
+		eventHolder.Event = p.processors.Run(eventHolder.Event)
+	}
+
+	var ok bool
+	if eventHolder.Event != nil {
+		//processor might decide to drop the event
+		ok = p.outlet.OnEvent(&eventHolder)
```

Comment:
Lets move the `ok = p.outlet.OnEvent(&eventHolder)` after the if / else and only have `if eventHolder.Event == nil { Bytes = 0} `. This should be the same logic. Then we also don't need the definition of ok.